### PR TITLE
CLN: replace %s syntax with .format in pandas.io.parsers

### PR DESCRIPTION
**Pull request by makeajourney**
progress towards #16130

- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by makeajourney** [comment]
The error occurred on python 2.7 from CI.
Is there any person who knows how I can run the test on python 2.7? I wanna know the environment setting.

**Comment by WillAyd** [comment]
@makeajourney can you merge master here?

**Comment by jreback** [comment]
can you merge master

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -2273,10 +2278,11 @@ def __init__(self, f, **kwds):
             raise ValueError('Only length-1 decimal markers supported')
 
         if self.thousands is None:
-            self.nonnum = re.compile('[^-^0-9^%s]+' % self.decimal)
+            self.nonnum = re.compile(
+                r'[^-^0-9^{decimal}]+'.format(decimal=self.decimal))
```
Comment (WillAyd) [inline pandas/io/parsers.py]: Hmm does converting this to a raw string when it wasn't previously affect the behavior at all? Might also be the root cause of your issue, though we technically drop Py2 support with v0.25 anyway

Comment (makeajourney) [reply pandas/io/parsers.py]: Temporarily, I want to try to put that code back for clearing the problem from ci.
And then I'll find out how can I solve the problem.

**Comment by WillAyd** [comment]
Can you merge master and check CI failures?

**Comment by makeajourney** [comment]
I tried to install the python 2.7 using the pyenv, but I failed. At the same time, I tried to install and used the anaconda2, but It was a bad idea. Anyway, I've been finding another way to test with the python2.7 environment. Maybe I can use the docker to test.

**Comment by WillAyd** [comment]
@makeajourney if you already have conda installed you shouldn't need a separate installation to get a Py27 environment up. 

https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-with-commands

```sh
conda create -n pandas-dev2 python=2.7
```

File: pandas/io/parsers.py (hunk 1 above)
Comment (makeajourney) [reply pandas/io/parsers.py]: This change has no problem :)

File: pandas/io/parsers.py [hunk 2]
```diff
@@ -2557,7 +2563,8 @@ def _infer_columns(self):
 
                         while cur_count > 0:
                             counts[col] = cur_count + 1
-                            col = "%s.%d" % (col, cur_count)
+                            col = '{column}.{count}'.format(
+                                column=u''.join(col), count=cur_count)
```
Comment (makeajourney) [inline pandas/io/parsers.py]: I think there is a problem with ASCII encoding by python 2.7 on Windows. I've tried to solve this problem by changing some codes. But it does not work.

**Comment by WillAyd** [comment]
OK no worries. We are going to drop Py27 supported with the 0.25 release anyway so I would think it's OK to ignore any Py27 failures for that release (unless @jreback objects).

**Comment by makeajourney** [comment]
Okay.
Then I'm going to drop some commits related to the above problems. That's not good to look.

**Comment by WillAyd** [comment]
@makeajourney we just dropped the Py27 CI jobs - want to merge master to give this another shot?

**Comment by makeajourney** [comment]
@WillAyd Thanks for letting me know :) I wish there's no more error 😁

**Comment by jreback** [comment]
pls merge master

**Comment by jreback** [comment]
thanks !

**Pull request merged**